microcosm_caching/decorators.py

```diff
@@ -252,3 +252,77 @@ def cache(*args, **kwargs) -> Schema:
 
         return cache
     return decorator
+
+
+def invalidate_batch(
+    component,
+    batch_attribute,
+    invalidations: List[Invalidation],
+    cache_prefix: Optional[str] = None,
+    lock_ttl=DEFAULT_LOCK_TTL,
+    schema_version: Optional[str] = None,
+):
+    """
+    Invalidates a set of prescribed keys, based on a combination of:
+        * specified arguments
+        * schema
+        * a "batch" attribute
+
+    This function is meant for cases of batch uploads, such that there would
+    exist a set of resources created as a part of this operation which may each
+    need to trigger their own invalidations. This is subject to the same limitations
+    as the `invalidates` decorator.
+
+    """
+    graph = component.graph
+    metrics = get_metrics(graph)
+    resource_cache: CacheBase = graph.resource_cache
+
+    version = schema_version or get_build_version(graph)
+    cache_prefix = cache_prefix or graph.metadata.name
+
+    def delete_from_cache(values) -> None:
```

Comment:
This looks pretty similar to the above decorator. I may want to extract this out eventually.

Reply:
agreed. but this is fine for now.